packages/transformers/webextension/src/WebExtensionTransformer.js

```diff
@@ -182,17 +182,20 @@ async function collectDependencies(
     }
   }
   if (program.web_accessible_resources) {
+    const rawExtSet = new Set(['.json', '.js']);
     let war = [];
     for (let i = 0; i < program.web_accessible_resources.length; ++i) {
       // TODO: this doesn't support Parcel resolution
       const currentEntry = program.web_accessible_resources[i];
       const files = isMV2 ? [currentEntry] : currentEntry.resources;
-      let currentFiles = [];
-      for (let j = 0; j < files.length; ++j) {
-        const globFiles = (
-          await glob(path.join(assetDir, files[j]), fs, {})
-        ).map(fp =>
-          asset.addURLDependency(path.relative(assetDir, fp), {
+      
+      const globFiles = await Promise.all(files.map(
+        file => glob(path.join(assetDir, file), fs, {})
+      )).then((filePaths) => {
+        filePaths.forEach(filePath => {
+          const extension = path.extname(filePath);
+          asset.addURLDependency(path.relative(assetDir, filePath), {
```

Comment:
We used `map` before for a reason: the return value of `asset.addURLDependency` is actually important. So probably switch back to that. Overall for consistency with our other code I would just do:

```js
const globFiles = await Promise.all(files.map(
  file => glob(path.join(assetDir, file), fs, {})
));
const currentFiles = globFiles.flatMap(fp =>
  asset.addURLDependency(path.relative(assetDir, fp), {
    pipeline: path.extname(fp) == '.json' ? 'raw' : undefined,
  })
);
```

And remove `rawExtSet`. I don't think `.js` files should remain untransformed, that doesn't seem like expected behavior (and I think it might break some built-in HMR stuff).

Reply:
BTW thanks so much for the code review, much much nicer :D

Reply:
BTW is the 

```
loc: {
            filePath,
            ...getJSONSourceLocation(
              ptrs[
                `/web_accessible_resources/${i}${
                  isMV2 ? '' : `/resources/${j}`
                }`
              ],
            ),  
          }
```
not needed at all?  Or what should I replace the `j` index with in the flatmap :d....

Reply:
Actually you're right, you should just revert to the old logic without `flatMap` or find some other way to preserve the proper `loc` handling.

Reply:
Make sense 👍